test/statistics_listener.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import sys
+from threading import Lock
+from typing import List
+
+from metrics_statistics_msgs.msg import MetricsMessage
+
+import rclpy
+from rclpy.node import Node
+from rclpy.task import Future
+
+# QoS history_depth to use for listener node
+QOS_DEPTH = 10
+
+"""
+Helper class for running system meterics end-to-end tests.
+
+This class is a Node that subscribes to the topic to which MetricsMessages are published
+and checks that all expected messages are received.
+"""
+
+class StatisticsListener(Node):
+    """Listen for MetricsMessages published on the /system_metrics topic."""
+
+    def __init__(self, future: Future, expected_lifecycle_nodes: List,
+                 expected_number_of_messages_to_receive: int, subscription_topic: str):
+        super().__init__('statisticsListener')
+
+        self.sub = self.create_subscription(MetricsMessage,
+                                            subscription_topic,
+                                            self.listener_callback,
+                                            QOS_DEPTH)
+        # set up logging
+        logger = logging.getLogger()
+        # setting to debug for end to end test soak
+        logger.setLevel(logging.DEBUG)
+
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setLevel(logging.DEBUG)
+        formatter = logging.Formatter('[%(module)s] [%(levelname)s] [%(asctime)s]: %(message)s')
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
```

Comment:
Why move the logging config into this class instead of leaving in a free function? Setting up the logger shouldn't be this class's responsibility. https://docs.python.org/3/howto/logging-cookbook.html

Reply:
Consolidated logger to just 1 free function.